src/NHSD.BuyingCatalogue.Ordering.Api/Controllers/OrderingPartyController.cs

```diff
@@ -74,18 +76,70 @@ public async Task<ActionResult<OrderingPartyModel>> GetAsync(string orderId)
         }
 
         [HttpPut]
-        public ActionResult Update(string orderId, OrderingPartyModel model)
+        public async Task<ActionResult> Update(string orderId, OrderingPartyModel model)
         {
-            if (orderId is null)
+            if (model is null)
+            {
+                throw new ArgumentNullException(nameof(model));
+            }
+
+            var order = await _orderRepository.GetOrderByIdAsync(orderId);
+            if (order is null)
             {
                 return NotFound();
             }
 
-            if (model is null)
+            var primaryOrganisationId = User.GetPrimaryOrganisationId();
+            if (primaryOrganisationId != order.OrganisationId)
             {
-                throw new ArgumentNullException(nameof(model));
+                return Forbid();
+            }
+
+            var contact = model.PrimaryContact;
+            var isContactValid = OrderPartyContact.Create(contact.FirstName, contact.LastName, contact.EmailAddress, contact.TelephoneNumber);
+
+            if (!isContactValid.IsSuccess)
+            {
+                return BadRequest(new ErrorsModel(isContactValid.Errors.Select(x => new ErrorModel(x.Id, x.Field))));
+            }
+
+            order.OrganisationContact = isContactValid.Value.GetContact();
+
+            var organisation = model.Organisation;
+            var addressModel = organisation.Address;
+            var address = new Address
+            {
+                Line1 = addressModel.Line1,
+                Line2 = addressModel.Line2,
+                Line3 = addressModel.Line3,
+                Line4 = addressModel.Line4,
+                Line5 = addressModel.Line5,
+                Town = addressModel.Town,
+                County = addressModel.County,
+                Postcode = addressModel.Postcode,
+                Country = addressModel.Country
+            };
+            var isOrganisationValid = OrderPartyOrganisation.Create(organisation.Name, organisation.OdsCode, address);
+
+            if (!isOrganisationValid.IsSuccess)
+            {
+                return BadRequest(new ErrorsModel(isContactValid.Errors.Select(x => new ErrorModel(x.Id, x.Field))));
             }
 
+            var updateOrganisation = isOrganisationValid.Value.GetOrganisation();
+
+            order.OrganisationName = updateOrganisation.Name;
+            order.OrganisationOdsCode = updateOrganisation.OdsCode;
+            order.OrganisationAddress = updateOrganisation.Adress;
+
+            var name = User.Identity.Name;
+
+            order.SetLastUpdatedByName(name);

```

Comment:
Might be a good idea to put all these three variables into the set method (on order)

So we dont have to set them each time (just call the SetLastUpdated) on order to set

LastUpdatedBy
LastUpdated
LastUodatedByName